### PR TITLE
9735 force using TLS 1.3 for NY-PHD

### PR DESCRIPTION
**Pull request by jalbinson**
This PR forces TLS v1.2 for Rest Transport that uses a client authentication certificate.  We do not want receivers to be able to negotiate to an outdated TLS version which was possible in the previous implementation.  This is currently only used with NY-PHD but could be others in the future.

Test Steps:
1. Pull this branch to your local machine
2. Update your locally running ReportStream with NY-PHD production settings and credentials
3. Use the check functionality in the org UI to test connecting to the NY-PHD receiver
4. The dialog will let you know if everything works

## Changes
- Update SSL version in Rest Transport

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes https://github.com/CDCgov/prime-reportstream/issues/9735

### CODE REVIEW

**Review by oslynn — APPROVED** [review]

File: prime-router/src/main/kotlin/transport/RESTTransport.kt [hunk 1]
```diff
@@ -500,7 +500,7 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
             keyManagerFactory.init(keyStore, jksPasscode)
             // create the sslContext, type TLS, with the keyManager
             // note the ktor sample code uses a TrustManger instead, but that fails with the NY cert file
-            val sslContext = SSLContext.getInstance("TLS")
+            val sslContext = SSLContext.getInstance("TLSv1.2")
```
Comment (snesm) [inline prime-router/src/main/kotlin/transport/RESTTransport.kt:503]: 1.3 would be a better long term choice. https://github.com/CDCgov/prime-reportstream/pull/9515/files

I'm told that this code affects all REST partners.

**Review by snesm — CHANGES_REQUESTED** [review]

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 1 above)
Comment (snesm) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:503]: and NY is capable of 1.3
https://www.ssllabs.com/ssltest/analyze.html?d=api2.health.ny.gov&hideResults=on

Comment (jalbinson) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:503]: It only affects NY as the `getSslContext` function only runs given the receiver has a configured JKS for client certificate auth. Our other REST partners (OK, WA) don't use that so this code does not run.

Comment (snesm) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:503]: In that case, this should be"TLSv1.3".

Comment (jalbinson) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:503]: Sounds good but let me double check that this works with JDK 11.

Comment (jalbinson) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt]: worked fine

Screenshot 2023-06-27 at 2 40 51 PM

**Review by snesm — APPROVED** [review]

**Pull request merged**